team_token_integration_test.go

```diff
@@ -39,6 +40,54 @@ func TestTeamTokensCreate(t *testing.T) {
 		assert.Equal(t, err, ErrInvalidTeamID)
 	})
 }
+
+func TestTeamTokens_CreateWithOptions(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	tmTest, tmTestCleanup := createTeam(t, client, nil)
+	defer tmTestCleanup()
+
+	var tmToken string
+	t.Run("with valid options", func(t *testing.T) {
+		tt, err := client.TeamTokens.CreateWithOptions(ctx, tmTest.ID, TeamTokenCreateOptions{})
+		require.NoError(t, err)
+		require.NotEmpty(t, tt.Token)
+		tmToken = tt.Token
+	})
+
+	t.Run("when a token already exists", func(t *testing.T) {
+		tt, err := client.TeamTokens.CreateWithOptions(ctx, tmTest.ID, TeamTokenCreateOptions{})
+		require.NoError(t, err)
+		require.NotEmpty(t, tt.Token)
+		assert.NotEqual(t, tmToken, tt.Token)
+	})
+
+	t.Run("without valid team ID", func(t *testing.T) {
+		tt, err := client.TeamTokens.CreateWithOptions(ctx, badIdentifier, TeamTokenCreateOptions{})
+		assert.Nil(t, tt)
+		assert.Equal(t, err, ErrInvalidTeamID)
+	})
+
+	t.Run("without an expiration date", func(t *testing.T) {
+		tt, err := client.TeamTokens.CreateWithOptions(ctx, tmTest.ID, TeamTokenCreateOptions{})
+		require.NoError(t, err)
+		require.NotEmpty(t, tt.Token)
+		tmToken = tt.Token
```

Comment:
💡 We could check here that the returned token doesnt have an expiration date, in order to explicitly assert that not passing an ExpiredAt option works as expected

`assert.Empty(t, tt.ExpiredAt)`